peekaboo/db.py

```diff
@@ -228,11 +189,9 @@ def analysis_save(self, sample):
                 result=sample.result,
                 reason=sample.reason)
 
-        analysis.sample = sample_info
-
         with self.__lock:
             session = self.__session()
-            session.add(analysis)
+            session.add(sample_info)
```

Comment:
I know this is only an interim step in a larger change, but this seems inefficient in the case of sample info already existing by saving/updating it again. Adding the a new entry should either move into the `if` or the if should go away right in this change (with a bit of a elaborative comment regarding the repercussions). The latter seems fair to me because we're preserving this part of the analysis journal by creating a new sample info for each analysis, admittedly without any way to distinguish them - until a later change in the series.
Alternatively we could invert the if and do a return if `sample_info` is not None and then create the new `sample_info` in the main code path of the method.